### PR TITLE
[LPT] Tests migration to API 2.0

### PR DESCRIPTION
**Pull request by eshoguli**
### Details:
 - *LPT Tests migration to API 2.0*

### Tickets:
 - *CVS-111374*

### CODE REVIEW

File: src/tests/functional/plugin/shared/src/low_precision_transformations/add_transformation.cpp [hunk 1]
```diff
@@ -58,6 +58,13 @@ void AddTransformation::SetUp() {
     AddTestValues param;
     std::tie(precision, inputShape, targetDevice, param) = this->GetParam();
 
+    ngraph::PartialShape inputShape2 = inputShape;
```
Comment (ilyachur) [inline src/tests/functional/plugin/shared/src/low_precision_transformations/add_transformation.cpp:61]:
```suggestion
    ov::PartialShape inputShape2 = inputShape;
```

File: src/tests/functional/plugin/shared/src/low_precision_transformations/concat_transformation.cpp [hunk 2]
```diff
@@ -34,15 +36,27 @@ std::string ConcatTransformation::getTestCaseName(const testing::TestParamInfo<C
...
 }
 
-InferenceEngine::Blob::Ptr ConcatTransformation::GenerateInput(const InferenceEngine::InputInfo &info) const {
-    ngraph::PartialShape inputShape;
-    ngraph::element::Type netPrecision;
-    std::string targetDevice;
-    ConcatTransformationTestValues testValues;
-    std::tie(netPrecision, inputShape, targetDevice, testValues) = this->GetParam();
+ov::test::utils::InputsMap ConcatTransformation::get_input_map() {
+    auto generate_default = [](const std::shared_ptr<ngraph::Node>&node,
```
Comment (ilyachur) [inline src/tests/functional/plugin/shared/src/low_precision_transformations/concat_transformation.cpp:40]:
```suggestion
    auto generate_default = [](const std::shared_ptr<ov::Node>&node,
```
Please grep for `ngraph` keyword in order to remove legacy namespace at all

File: src/tests/functional/plugin/shared/src/low_precision_transformations/mat_mul_transformation.cpp [hunk 3]
```diff
@@ -13,6 +13,7 @@
 
 #include "ngraph/op/op.hpp"
```
Comment (ilyachur) [inline src/tests/functional/plugin/shared/src/low_precision_transformations/mat_mul_transformation.cpp:14]: Please remove all legacy headers!!!

File: src/tests/functional/plugin/shared/src/low_precision_transformations/mat_mul_transformation.cpp [hunk 4]
```diff
@@ -21,10 +22,9 @@ namespace LayerTestsDefinitions {
 
 std::string MatMulTransformation::getTestCaseName(const testing::TestParamInfo<MatMulTransformationParams>& obj) {
     ngraph::element::Type precision;
-    ngraph::PartialShape inputShape;
```
Comment (ilyachur) [inline src/tests/functional/plugin/shared/src/low_precision_transformations/mat_mul_transformation.cpp:24]: Why did you remove input shape for these tests?
Is it related with these changes? `ngraph::PartialShape` can be changed to `ov::PartialShape`

File: src/tests/functional/plugin/shared/src/low_precision_transformations/mat_mul_transformation.cpp [hunk 5]
```diff
@@ -38,31 +38,39 @@ std::string MatMulTransformation::getTestCaseName(const testing::TestParamInfo<M
...
+    };
+
+    static ov::test::utils::InputsMap inputs_map{
+        { ov::op::Op::get_type_info_static(), generate_default }
+    };
+    return inputs_map;
 }
 
 void MatMulTransformation::SetUp() {
     ngraph::element::Type precision;
```
Comment (ilyachur) [inline src/tests/functional/plugin/shared/src/low_precision_transformations/mat_mul_transformation.cpp:69]: Also please remove ngraph namespace

File: src/tests/functional/plugin/shared/src/low_precision_transformations/output_layers_handling_in_transformations.cpp [hunk 6]
```diff
@@ -12,6 +12,7 @@
 #include <ie_core.hpp>
```
Comment (ilyachur) [inline src/tests/functional/plugin/shared/src/low_precision_transformations/output_layers_handling_in_transformations.cpp:12]: Please remove legacy IE headers too

File: src/tests/functional/plugin/shared/src/low_precision_transformations/reduce_min_transformation.cpp [hunk 7]
```diff
@@ -7,7 +7,7 @@
 #include <string>
 #include <vector>
 #include <ngraph/ngraph.hpp>
```
Comment (ilyachur) [inline src/tests/functional/plugin/shared/src/low_precision_transformations/reduce_min_transformation.cpp:9]: Do not use global headers.

File: src/tests/functional/plugin/shared/include/low_precision_transformations/depth_to_space_transformation.hpp [hunk 8]
```diff
@@ -6,7 +6,7 @@
 
 #include <string>
 #include <memory>
-
+#include "ngraph/opsets/opset1.hpp"
```
Comment (nshchego) [inline src/tests/functional/plugin/shared/include/low_precision_transformations/depth_to_space_transformation.hpp:9]: `ngraph/opsets/opset1.hpp` -> `openvino/op/depth_to_space.hpp`

File: src/tests/functional/plugin/shared/src/low_precision_transformations/pull_reshape_through_dequantization_transformation.cpp [hunk 9]
```diff
@@ -42,7 +42,7 @@ std::string PullReshapeThroughDequantizationTransformation::getTestCaseName(cons
 }
 
 void PullReshapeThroughDequantizationTransformation::SetUp() {
-    // threshold = 0.1f;
+    // rel_threshold = 0.1f;
```
Comment (nshchego) [inline src/tests/functional/plugin/shared/src/low_precision_transformations/pull_reshape_through_dequantization_transformation.cpp:45]: Please remove commented code.

File: src/tests/functional/plugin/shared/src/low_precision_transformations/reduce_mean_transformation.cpp [hunk 10]
```diff
@@ -7,7 +7,7 @@
 #include <string>
 #include <vector>
 #include <ngraph/ngraph.hpp>
-
+#include <ngraph/opsets/opset1.hpp>
```
Comment (nshchego) [inline src/tests/functional/plugin/shared/src/low_precision_transformations/reduce_mean_transformation.cpp:10]: `ngraph/opsets/opset1.hpp` -> `openvino/op/reduce_mean.hpp`

File: src/tests/functional/plugin/shared/include/low_precision_transformations/concat_transformation.hpp [hunk 11]
```diff
@@ -7,6 +7,8 @@
 #include <string>
 #include <memory>
 
+#include "ngraph/opsets/opset1.hpp"
```
Comment (iefode) [inline src/tests/functional/plugin/shared/include/low_precision_transformations/concat_transformation.hpp:10]: why `ngraph`?

File: src/tests/functional/plugin/shared/include/low_precision_transformations/concat_transformation.hpp [hunk 12]
```diff
@@ -34,7 +36,7 @@ class ConcatTransformation :
     public LayerTestsUtils::LayerTransformation {
 public:
     static std::string getTestCaseName(const testing::TestParamInfo<ConcatTransformationParams>& obj);
-    InferenceEngine::Blob::Ptr GenerateInput(const InferenceEngine::InputInfo &info) const override;
+    ov::test::utils::InputsMap get_input_map() override;
```
Comment (iefode) [inline src/tests/functional/plugin/shared/include/low_precision_transformations/concat_transformation.hpp:39]: there is `generate_inputs` method. Also you can use basic method to generate correct data per op (please check the implementation). I believe we should use default way to generate input data... (we use the same approach in conformance)
Could you please share the details why you decide to use `get_input_map()`???

File: src/tests/functional/plugin/shared/src/low_precision_transformations/concat_transformation.cpp [hunk 13]
```diff
@@ -51,6 +65,15 @@ void ConcatTransformation::SetUp() {
     ConcatTransformationTestValues testValues;
     std::tie(precision, inputShape, targetDevice, testValues) = this->GetParam();
 
+    std::vector<ngraph::PartialShape> inputs;
```
Comment (iefode) [inline src/tests/functional/plugin/shared/src/low_precision_transformations/concat_transformation.cpp:68]: `ngraph` -> `ov`

File: src/tests/functional/plugin/shared/src/low_precision_transformations/concat_with_intermediate_transformation.cpp [hunk 14]
```diff
@@ -37,17 +39,25 @@ std::string ConcatWithIntermediateTransformation::getTestCaseName(const testing:
...
-InferenceEngine::Blob::Ptr ConcatWithIntermediateTransformation::GenerateInput(const InferenceEngine::InputInfo &info) const {
-    ngraph::element::Type netPrecision;
-    ngraph::PartialShape inputShape;
-    std::string targetDevice;
-    ngraph::pass::low_precision::LayerTransformation::Params trasformationParams;
-    bool transparentIntermediate;
-    bool multichannel;
-    std::tie(netPrecision, inputShape, targetDevice, trasformationParams, transparentIntermediate, multichannel) = this->GetParam();
+ov::test::utils::InputsMap ConcatWithIntermediateTransformation::get_input_map() {
+    auto generate_default = [](const std::shared_ptr<ngraph::Node>& node,
```
Comment (iefode) [inline src/tests/functional/plugin/shared/src/low_precision_transformations/concat_with_intermediate_transformation.cpp:43]: I saw the code duplications. Also I have left one comment connected to this method... Let's discuss offline to get more info and choose the best techical solution

File: src/tests/functional/plugin/shared/src/low_precision_transformations/concat_with_intermediate_transformation.cpp [hunk 15]
```diff
@@ -66,6 +76,18 @@ void ConcatWithIntermediateTransformation::SetUp() {
     bool multichannel;
     std::tie(ngPrecision, inputShape, targetDevice, trasformationParams, transparentIntermediate, multichannel) = this->GetParam();
 
+    ngraph::PartialShape inputShape1 = inputShape;
```
Comment (iefode) [inline src/tests/functional/plugin/shared/src/low_precision_transformations/concat_with_intermediate_transformation.cpp:79]: Please, do not use outdated namespaces! `ngraph` -> `ov`

File: src/tests/functional/plugin/shared/src/low_precision_transformations/convolution_transformation.cpp [hunk 16]
```diff
@@ -53,8 +55,8 @@ void ConvolutionTransformation::SetUp() {
         param.fakeQuantizeOnWeights);
 }
 
-void ConvolutionTransformation::Run() {
-    LayerTestsCommon::Run();
+void ConvolutionTransformation::run() {
+    LayerTransformation::run();
```
Comment (iefode) [inline src/tests/functional/plugin/shared/src/low_precision_transformations/convolution_transformation.cpp:59]: Open question: Replace by new namespace?? `ov::Transformation` or something else? Where is method definition?

File: src/tests/functional/shared_test_classes/include/shared_test_classes/base/low_precision_transformations/layer_transformation.hpp [hunk 17]
```diff
@@ -30,16 +30,16 @@ class LayerTransformationParamsNGraphFactory {
...
     LayerTransformation();
 
-    static InferenceEngine::Blob::Ptr GenerateInput(
-        const ngraph::element::Type precision,
-        const InferenceEngine::TensorDesc& tensorDesc,
-        const float k = 1.f);
+    ov::test::utils::InputsMap get_input_map() override;
 
-    static std::pair<float, float> getQuantizationInterval(const ngraph::element::Type precision);
+    ov::test::utils::CompareMap get_compare_map() override;
```
Comment (iefode) [inline src/tests/functional/shared_test_classes/include/shared_test_classes/base/low_precision_transformations/layer_transformation.hpp:42]: why we cannot use the default way?

File: src/tests/functional/shared_test_classes/include/shared_test_classes/base/ov_subgraph.hpp [hunk 18]
```diff
@@ -35,6 +37,8 @@ class SubgraphBaseTest : public ov::test::TestsCommon {
     virtual void compile_model();
     virtual void init_ref_function(std::shared_ptr<ov::Model> &funcRef, const std::vector<ov::Shape>& targetInputStaticShapes);
     virtual void generate_inputs(const std::vector<ov::Shape>& targetInputStaticShapes);
+    virtual ov::test::utils::InputsMap get_input_map();
```
Comment (iefode) [inline src/tests/functional/shared_test_classes/include/shared_test_classes/base/ov_subgraph.hpp:40]: I think that is extra code (we use these map to generate and compare data. Also they are defined out of the class)... Let's discuss offline.

File: src/tests/functional/shared_test_classes/include/shared_test_classes/base/utils/ranges.hpp [hunk 19]
```diff
@@ -60,7 +60,7 @@ struct InputGenerateData {
             auto max_ref = ConstRanges::max;
             if (min_orig < min_ref || min_orig == 0)
                 start_from = min_ref;
-            range = (max_orig > max_ref || max_orig == 10 ? max_ref : max_orig - start_from) - start_from;
+            range = static_cast<uint32_t>((max_orig > max_ref || max_orig == 10 ? max_ref : max_orig - start_from) - start_from);
```
Comment (iefode) [inline src/tests/functional/shared_test_classes/include/shared_test_classes/base/utils/ranges.hpp:63]: not required. It will be casted by compiler automatically

File: src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp [hunk 20]
```diff
@@ -170,7 +169,7 @@ void SubgraphBaseTest::compare(const std::vector<ov::Tensor>& expected,
                     inputNode = nextNodePtr;
                 }
             }
-            auto it = compareMap.find(inputNode->get_type_info());
+            auto it = compareMap.size() == 1 ? compareMap.begin() : compareMap.find(inputNode->get_type_info());
```
Comment (iefode) [inline src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp:172]: HARDCODE!!!! What is that?! I have left the comment connected to these method above. Let's setup short call

File: src/tests/functional/plugin/shared/src/low_precision_transformations/convolution_transformation.cpp (hunk 16 above)
Comment (eshoguli) [reply src/tests/functional/plugin/shared/src/low_precision_transformations/convolution_transformation.cpp:59]: it's not namespace. `LayerTransformation` is base class for all LPT functional tests.

File: src/tests/functional/shared_test_classes/include/shared_test_classes/base/low_precision_transformations/layer_transformation.hpp (hunk 17 above)
Comment (eshoguli) [reply src/tests/functional/shared_test_classes/include/shared_test_classes/base/low_precision_transformations/layer_transformation.hpp:42]: The previous version of `default way` is used. It's not my implementation.
We need additional time to use new comparation logic. Anyway, It's not part of migration to OpenVINO API 2.0.

File: src/tests/functional/shared_test_classes/include/shared_test_classes/base/utils/ranges.hpp (hunk 19 above)
Comment (eshoguli) [reply src/tests/functional/shared_test_classes/include/shared_test_classes/base/utils/ranges.hpp:63]: not will be casted in Windows if all warnings as error

File: src/tests/functional/plugin/shared/include/low_precision_transformations/concat_transformation.hpp (hunk 12 above)
Comment (iefode) [reply src/tests/functional/plugin/shared/include/low_precision_transformations/concat_transformation.hpp:39]: Discussed offline. This method should be removed. We will use static ranges for fake quintize in `generate_input`

**Comment by ilyachur** [comment]
@eshoguli Do you have an update for this PR?

**Comment by iefode** [comment]
@eshoguli Do you have a plan to apply review comments?

**Comment by ilyachur** [comment]
@eshoguli Am I right that right now you don't have any plans to fix comments and complete this PR?
In this case, we have a risk that LPT tests will be a show stopper for removing legacy API.
